docs/modules/SPEC.md

```diff
@@ -0,0 +1,80 @@
+# Module Specification
+
+This document outlines the recommended structure of Cosmos SDK modules. These
+ideas are meant to be applied as suggestions. Application developers are encouraged
+to improve upon and contribute to module structure and development design.
+
+## Structure
+
+A typical Cosmos SDK module can be structured as follows:
+
+```shell
+x/{module}
+├── client
```

Comment:
How about flatten `client` further? I mean: instead of having `cli` and `rest`  as subpackages, how about promoting them to live module top level and get rid of client?

Reply:
Hmmm, I don't this that is a wise idea. They're both clients and should be grouped as such.